src/Application/Actions/RegularGivingMandate/Get.php

```diff
@@ -36,19 +36,24 @@ protected function action(Request $request, Response $response, array $args): Re
         if (! $this->environment->isFeatureEnabledRegularGiving()) {
             throw new HttpNotFoundException($request);
         }
+
+        Assertion::keyExists($args, "mandateId");
+        $mandateId = $args["mandateId"];
+        Assertion::string($mandateId);
         if (empty($args['mandateId'])) {
-            throw new DomainRecordNotFoundException('Missing donation ID');
+            throw new DomainRecordNotFoundException('Missing mandate ID ' . $mandateId);
         }
 
+        // @todo-regular-giving: make sure that the donor requesting the mandate is the owner of the mandate
         $donorId = $request->getAttribute(PersonWithPasswordAuthMiddleware::PERSON_ID_ATTRIBUTE_NAME);
         \assert($donorId instanceof PersonId);
         $uuid = Uuid::fromString((string) $args['mandateId']);
         $mandate = $this->regularGivingMandateRepository->findOneByUuid($uuid);
-
```

Comment:
Cannot throw an exception here because sa:check says $mandate is never falsy, but we are throwing exception up a level in the `regularGivingMandateRepository->findOneByUuid($uuid)` when the mandate wasn't found

Reply:
@dorota-joanna that's probably because you've asserted that the mandate is not null - if you assert it Psalm will believe you.